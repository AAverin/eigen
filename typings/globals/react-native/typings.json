{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
