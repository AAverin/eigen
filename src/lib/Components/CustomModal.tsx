import { theme } from "lib/Components/Bidding/Elements/Theme"
import React from "react"
import { Modal, TouchableWithoutFeedback, View, ViewProperties } from "react-native"
import { Sans14, SansMedium14 } from "./Bidding/Elements/Typography"
import { SecondaryOutlineButton } from "./Buttons"

interface CustomModalProps extends ViewProperties {
  headerText: string
  detailText: string
  visible?: boolean
  closeModal?: () => void
}

export class CustomModal extends React.Component<CustomModalProps, any> {
  constructor(props) {
    super(props)

    this.state = { modalVisible: props.visible || false }
  }

  componentDidUpdate(prevProps) {
    if (this.props.visible !== prevProps.visible) {
      this.setState({ modalVisible: this.props.visible })
    }
  }

  closeModal() {
    if (this.props.closeModal) {
      this.props.closeModal()
    }
    this.setState({ modalVisible: false })
  }

  render() {
    const { headerText, detailText } = this.props

    return (
      <View style={{ marginTop: 22 }}>
        <Modal animationType="fade" transparent={true} visible={this.state.modalVisible}>
          <TouchableWithoutFeedback onPress={() => this.closeModal()}>
            <View
              style={{
                backgroundColor: "#00000099",
                flex: 1,
                flexDirection: "column",
                justifyContent: "center",
                alignItems: "center",
              }}
            >
              <TouchableWithoutFeedback onPress={null}>
                <View style={{ width: 300, backgroundColor: "white", padding: 20, opacity: 1, borderRadius: 2 }}>
                  <View style={{ paddingBottom: 20 }}>
                    <SansMedium14>{headerText}</SansMedium14>
                  </View>
                  <View style={{ paddingBottom: 30 }}>
                    <Sans14 color={theme.colors.black60}>{detailText}</Sans14>
                  </View>
                  <SecondaryOutlineButton
                    text="Ok"
                    style={{ height: 40, borderRadius: 2, borderWidth: 2 }}
                    onPress={() => {
                      this.closeModal()
                    }}
                  />
                </View>
              </TouchableWithoutFeedback>
            </View>
          </TouchableWithoutFeedback>
        </Modal>
      </View>
    )
  }
}
