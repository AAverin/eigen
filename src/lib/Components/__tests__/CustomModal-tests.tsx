import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import { CustomModal } from "../CustomModal"

it("looks like expected", () => {
  const tree = renderer
    .create(
      // tslint:disable-next-line:no-empty
      <CustomModal headerText="An error occurred" detailText="This is an error moop." />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
