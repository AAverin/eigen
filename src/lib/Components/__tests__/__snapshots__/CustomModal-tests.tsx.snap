// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks like expected 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000099",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 2,
            "opacity": 1,
            "padding": 20,
            "width": 300,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "paddingBottom": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={2}
            lineHeight={5}
            style={
              Array [
                Object {
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 16,
                  "lineHeight": 5,
                },
                undefined,
              ]
            }
          >
            An error occurred
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            color="#666"
            ellipsizeMode="tail"
            fontSize={2}
            lineHeight={5}
            style={
              Array [
                Object {
                  "color": "#666",
                  "fontFamily": "Unica77LL-Regular",
                  "fontSize": 16,
                  "lineHeight": 5,
                },
                undefined,
              ]
            }
          >
            This is an error moop.
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderColor": "rgba(229, 229, 229, 1)",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AvantGardeGothicITC",
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 14,
                  "opacity": 1,
                },
              ]
            }
          >
            Ok
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
